### PR TITLE
Added a factory for smallgroups, cleaned up some of the frontend code

### LINKED ISSUE #556: could someone fix delete day on smallgroup?
it has stopped working for a few weeks.

### PR DESCRIPTION
**Pull request by agundy**
fixed #556

### CODE REVIEW

**Comment by kmcnellis** [comment]
Looks like you're failing attendance tests

File: client/components/viewAttendance/viewAttendance.directive.js [hunk 1]
```diff
@@ -90,34 +90,67 @@ angular.module('observatory3App')
...
-            $scope.showDayCode = true;
-          }
-          $scope.hide = true;
-          updateGroup();
-        });
+        if ($scope.endpoint === 'smallgroup') {
+          Smallgroup.createDaycode({
+            '_id': $scope.group._id,
+            'bonusDay': bonusDay ? true : false
+          }, function (data) {
```
Comment (kmcnellis) [inline client/components/viewAttendance/viewAttendance.directive.js]: Can we pull this out as a function so we don't have duplicate code?

File: server/api/attendance/attendance.controller.js [hunk 2]
```diff
@@ -314,6 +314,7 @@ exports.attend = function(req,res){
   if (!code) {return res.status(400).json('No Code Submitted');}
   // Uppercase code from client so it is case-insensitive. This must happen
   // after the above check, otherwise toUpperCase() might not exist.
+  console.log(code);
```
Comment (kmcnellis) [inline server/api/attendance/attendance.controller.js]: console.log

File: server/api/smallgroup/index.js [hunk 3]
```diff
@@ -12,8 +12,8 @@ router.get('/', auth.hasRole('mentor'), controller.index);
 router.post('/', auth.hasRole('mentor'), controller.create);
 
 // Generate a daycode or return the current day code for your smallgroup
-router.post('/daycode', auth.hasRole('mentor'), controller.daycode);
-router.delete('/day/:dayCode', auth.hasRole('mentor'), controller.deleteDay);
+router.post('/:id/daycode', auth.hasRole('mentor'), controller.daycode);
```
Comment (kmcnellis) [inline server/api/smallgroup/index.js]: Why do this? The factory can set the id parameter to be daycode and use the original endpoint.  Mentors should only be generating daycodes for their own small group anyways.

**Comment by kmcnellis** [comment]
@agundy ?

**Comment by agundy** [comment]
Will update later today hopefully! Thanks for the ping

**Comment by kmcnellis** [comment]
:(

File: server/api/smallgroup/index.js (hunk 3 above)
Comment (agundy) [inline server/api/smallgroup/index.js]: daycodes aren't guaranteed to be unique right now and therefore cannot be relied on for things like this

Comment (agundy) [inline server/api/smallgroup/index.js]: If you do by code you will delete all randomly generated attendance codes in the db. To stay consistent I'm pulling from just daycodes from the small group

**Comment by agundy** [comment]
Fixed for you @kmcnellis

**Comment by kmcnellis** [comment]
Thanks for the update!

**Comment by kmcnellis** [comment]
ahh.... Too many buttons :/

**Pull request merged**